src/db-adapters/Mongoose/MongooseAdapter.js

```diff
@@ -614,4 +596,29 @@ export default class MongooseAdapter {
 
     return words.join(" ");
   }
+
+  static getIdQueryType(idOrIds) {
+    const mode = typeof idOrIds === "string" ? "findOne" : "find";
+    let idQuery;
+
+    if (typeof idOrIds === "string") {
+      if (!this.idIsValid(idOrIds)) {
+        throw new APIError(404, undefined, "No matching resource found.", "Invalid ID.");
+      }
+
+      idQuery = {_id: idOrIds};
+    } else if (Array.isArray(idOrIds)) {
+      if (idOrIds.some(this.idIsValid)) {
```

Comment:
Right, I meant `.every()` with a `!` in front, as in my [prior comment](https://github.com/ethanresnick/json-api/pull/58#discussion_r38268692). That is, "throw if it's not the case that every id is valid", which is different from "throw if some ids are valid".